WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -0,0 +1,685 @@
+    override internal func lastSyncDate() -> NSDate? {
+        return blog?.lastPostsSync
+    }
+    
+    // MARK: - Actions
+    
+    @IBAction func handleAuthorFilterChanged(sender: AnyObject) {
+        if authorFilterSegmentedControl.selectedSegmentIndex == (Int) (PostAuthorFilter.Mine.rawValue) {
+            setCurrentPostAuthorFilter(.Mine)
+        } else {
+            setCurrentPostAuthorFilter(.Everyone)
+        }
+    }
+    
+    // MARK: - TableViewHandler
+    
+    private func entityName() -> String {
+        return String(Post.self)
+    }
+    
+    private func predicateForFetchRequest() -> NSPredicate {
+        var predicates = [NSPredicate]()
+        
+        if let blog = blog {
+            let basePredicate = NSPredicate(format: "blog = %@ && revision = nil", blog)
+            predicates.append(basePredicate)
+        }
+        
+        let typePredicate = NSPredicate(format: "postType = %@", postTypeToSync())
+        predicates.append(typePredicate)
+        
+        let searchText = currentSearchTerm()
+        var filterPredicate = currentPostListFilter()?.predicateForFetchRequest
+        
+        // If we have recently trashed posts, create an OR predicate to find posts matching the filter,
+        // or posts that were recently deleted.
+        if let recentlyTrashedPostObjectIDs = recentlyTrashedPostObjectIDs
+            where searchText?.characters.count == 0 && recentlyTrashedPostObjectIDs.count > 0 {
+            
+            let trashedPredicate = NSPredicate(format: "SELF IN %@", recentlyTrashedPostObjectIDs)
+            
+            if let originalFilterPredicate = filterPredicate {
+                filterPredicate = NSCompoundPredicate(orPredicateWithSubpredicates: [originalFilterPredicate, trashedPredicate])
+            } else {
+                filterPredicate = trashedPredicate
+            }
+        }
+        
+        if let filterPredicate = filterPredicate {
+            predicates.append(filterPredicate)
+        }
+        
+        if shouldShowOnlyMyPosts() {
+            let myAuthorID = blog?.account.userID ?? 0
+            
+            // Brand new local drafts have an authorID of 0.
+            let authorPredicate = NSPredicate(format: "authorID = %@ || authorID = 0", myAuthorID)
+            predicates.append(authorPredicate)
+        }
+        
+        if let searchText = searchText where searchText.characters.count > 0 {
+            let searchPredicate = NSPredicate(format: "postTitle CONTAINS[cd] %@", searchText)
+            predicates.append(searchPredicate)
+        }
+        
+        let predicate = NSCompoundPredicate(andPredicateWithSubpredicates: predicates)
+        return predicate
+    }
+    
+    // MARK: - Table View Handling
+    
+    private func tableView(tableView: UITableView, estimatedHeightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as! Post
```

Comment:
Following up on [this comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/5156#discussion_r60729344), I'm adding an assertion, a log message and returning zero here.

In any case this will probably trigger a cell request which will cause a program interruption.